tests/server/api/test_folders.py

```diff
@@ -0,0 +1,52 @@
+# coding=utf-8
+"""
+Unit tests for ``octoprint.server.api`` folders.
+"""
+
+from __future__ import absolute_import
+
+__author__ = "Gina Häußge <osd@foosel.net>"
+__license__ = 'GNU Affero General Public License http://www.gnu.org/licenses/agpl.html'
+__copyright__ = "Copyright (C) 2016 The OctoPrint Project - Released under terms of the AGPLv3 License"
+
+
+import json
+from mock import MagicMock
+import psutil
+from random import randint
+import unittest
+
+from octoprint.settings import settings
+from octoprint.server import app
+from octoprint.server.api import api
+
+# Initialize settings
+settings(init=True)
```

Comment:
This way you are using the actual settings (and if they don't exist yet, have new ones created, cluttering up whatever machine is running the test). You should instead `mock` the settings API and inject it into the resource under test. 

An example on how to that can be found [here](https://github.com/foosel/OctoPrint/blob/112a8b9c8bf7f5895f6a29dbe3d98ea4d9e0f400/tests/timelapse/test_timelapse_helpers.py#L23). In a nutshell:

  * in `setUp` create a patcher, getter and mock:

    ``` python
    def setUp(self):
        self.settings_patcher = mock.patch("octoprint.server.api.folders.settings") # needs to be adjusted to import
        self.settings_getter = self.settings_patcher.start()
        self.settings = mock.create_autospec(octoprint.settings.Settings)
        self.settings_getter.return_value = self.settings
    ```
  * in `cleanUp` clean up the patcher

    ``` python
    def cleanUp(self):
        self.settings_patcher.stop()
    ```
  * in your test method, mock out the behaviour you need from the settings in your expected scenario, e.g.

    ``` python
    def test_read_usage_for_folder(self):
        # setup
        self.settings.getBaseFolder.return_value = "/path/to/basefolder"

        # implement your test ...
    ```

Reply:
This is a big help, thank you! I will get this in there.